tests/gold_tests/autest-site/httpbin.test.ext

```diff
@@ -19,17 +19,21 @@
 from ports import get_port
 
 
-def MakeHttpBinServer(obj, name, port=False, ip=False, delay=False, public_ip=False, ssl=False, options={}):
-    data_dir = os.path.join(obj.RunDirectory, name)
+def MakeHttpBinServer(self, name, port=False, ip=False, delay=False, public_ip=False, ssl=False, options={}):
+    data_dir = os.path.join(self.RunDirectory, name)
     # create Process
-    p = obj.Processes.Process(name)
+    p = self.Processes.Process(name)
     if (port == False):
         port = get_port(p, "Port")
     if (ip == False):
         ip = '127.0.0.1'
     if (delay == False):
         delay = 0
 
+    self._RootRunable.SkipUnless(
+        Condition.HasProgram("gunicorn", "gunicorn needs be installed with httpbin package for this extension to run")
+    )
```

Comment:
This is great!

I would also add at the bottom ( I cannot seem to comment on this)

```
ExtendTest(MakeHttpBinServer, name="MakeHttpBinServer")
**ExtendTestRun(MakeHttpBinServer, name="MakeHttpBinServer")**
```
This will allow people to create a server at a TestRun scope, vs only the Test scope. I have a PR where I am adding this to the microserver and trafficserver extensions ( I need to push it still).

Reply:
Sure